bin/setup.sh

```diff
@@ -6,27 +6,53 @@
 #
 set -euxo pipefail
 
-# NOTE: this is written under the assumption that it will be built in canon
-sudo apt -y update && sudo apt -y upgrade && sudo apt install -y cmake python3.11 python3.11-venv wget
+if [[ ${OS} == "darwin" ]]; then
+    echo "Detected MacOS ${ARCH}"
+
+  if ! command -v brew >/dev/null; then
+     echo "Brew not installed. Please install brew!"
+     exit 1
+  fi
+  # Install required tools
+  brew install cmake python@3.11 wget unzip || true
```

Comment:
there is some dependency that is not installed here which is necessary to build the project. That needs to be added here.